adapter/internal/synchronizer/apis_fetcher.go

```diff
@@ -293,6 +294,7 @@ func FetchAPIsFromControlPlane(updatedAPIID string, updatedEnvs []string) {
 			break
 		} else if data.ErrorCode >= 400 && data.ErrorCode < 500 {
 			logger.LoggerSync.Errorf("Error occurred when retrieving APIs from control plane: %v", data.Err)
+			health.SetControlPlaneRestAPIStatus(false)
```

Comment:
Same concern mentioned in https://github.com/wso2/product-microgateway/pull/2184/files#r680110339

Reply:
https://github.com/wso2/product-microgateway/pull/2184#discussion_r680409762